File: LeetCode/02399. Check Distances Between Same Letters/02399.c
// https://leetcode.com/problems/check-distances-between-same-letters

bool checkDistances(char * s, int *d, int distSize) {
    int i, j, dist, len;

    for (i = 0, len = strlen(s); i < len; ++i) {
        dist = d[s[i]-'a'];

        if ((i - dist - 1 >= 0 && s[i - dist - 1] == s[i]) || (i + dist + 1 < len && s[i + dist + 1] == s[i])) {
            ;
        } else {
            return false;
        }
    }

    return true;
}

File: LeetCode/02399. Check Distances Between Same Letters/02399.go
// https://leetcode.com/problems/check-distances-between-same-letters

func checkDistances(s string, distance []int) bool {
    size, pair := len(s), len(s) >> 1

    for i := range s {
        idx := i + distance[s[i]-'a'] + 1
        if idx < size && s[i] == s[idx] {
            pair--
        }
    }

    return pair == 0
}
